builtin/aws/ecs/plugin.proto

```diff
@@ -4,16 +4,116 @@ package ecs;
 
 option go_package = "waypoint/builtin/aws/ecs";
 
+import "google/protobuf/any.proto";
+
 message Deployment {
   string url = 1;
   string task_arn = 2;
   string service_arn = 3;
   string target_group_arn = 4;
   string load_balancer_arn = 5;
   string cluster = 6;
+  google.protobuf.Any resource_state = 7;
 }
 
 message Release {
   string url = 1;
   string load_balancer_arn = 2;
 }
+
+// Resource contains the internal resource states.
+message Resource {
+  message Cluster {
+    string name = 1;
+    string arn = 2;
+  }
+
+  message ExecutionRole {
+    string name = 1;
+    string arn = 2;
+
+    // Indicates if an execution role was created by waypoint, and should be destroyed by waypoint
+    bool managed = 3;
+  }
+
+  message TaskRole {
+    string name = 1;
+    string arn = 2;
+  }
+
+  message Service {
+    string name = 1;
+    string arn = 2;
+    string cluster = 3;
+  }
+
+  message TargetGroup {
+    string name = 1;
+    string arn = 2;
+    int64 port = 3;
+  }
+
+  message LogGroup {
+    string name = 1;
+    string arn = 2;
+  }
+
+  message SecurityGroup {
+    string name = 1;
+    string id = 2;
+
+    // Indicates if a security group was created by waypoint, and should be destroyed by waypoint
+    bool managed = 3;
```

Comment:
We're only setting (not using) this for now, but it'll be valuable in the future when we decide to implement delete on security groups.